xarray/tests/test_treenode.py

```diff
@@ -224,6 +230,56 @@ def test_overwrite_child(self):
         assert marys_evil_twin.parent is john
 
 
+class TestChildren:
```

Comment:
It's a bit awkward to know whether to put tests for this class in `test_treenode.py` or `test_datatree.py`. 

On the one hand the concept of access to `.children` is very much part of the `TreeNode` structure 
 - it has nothing to do with `.data`. On the other hand it's hard to properly test it without full features that are only available on `DataTree`, such as `.copy`, a nice `repr`, and assertions. It's also a fully public-facing class, unlike `TreeNode`.

Reply:
This is a lot less awkward after #9482.